src/ntfs/read.rs

```diff
@@ -31,90 +31,221 @@ use transit_model_collection::*;
     fn from(stop: Stop) -> StopArea {
+        if stop.name.is_empty() {
+            warn!("stop_id: {}: for station stop_name is required", stop.id);
+        }
+        let coord = Coord::from((stop.lon, stop.lat));
+        if coord == Coord::default() {
+            warn!("stop_id: {}: for station coordinates are required", stop.id);
+        }
         StopArea {
             id: stop.id,
             name: stop.name,
             codes: KeysValues::default(),
             object_properties: KeysValues::default(),
             comment_links: CommentLinksT::default(),
             visible: stop.visible,
-            coord: Coord {
-                lon: stop.lon,
-                lat: stop.lat,
-            },
+            coord: coord,
             timezone: stop.timezone,
             geometry_id: stop.geometry_id,
             equipment_id: stop.equipment_id,
+            level_id: stop.level_id,
+        }
+    }
+}
+
+impl Into<Result<StopArea>> for Stop {
+    fn into(self) -> Result<StopArea> {
+        if self.name.is_empty() {
+            warn!("stop_id: {}: for platform stop_name is required", self.id);
+        }
+
+        let coord = Coord::from((self.lon, self.lat));
+        if coord == Coord::default() {
+            warn!(
+                "stop_id: {}: for platform coordinates are required",
+                self.id
+            );
         }
+        let stop_area = StopArea {
+            id: self.id,
+            name: self.name,
+            codes: KeysValues::default(),
+            object_properties: KeysValues::default(),
+            comment_links: CommentLinksT::default(),
+            visible: self.visible,
+            coord: coord,
+            timezone: self.timezone,
+            geometry_id: self.geometry_id,
+            equipment_id: self.equipment_id,
+            level_id: self.level_id,
+        };
+        Ok(stop_area)
     }
 }
-impl StopPoint {
-    fn from_with_type(stop: Stop, stop_type: StopType) -> StopPoint {
+
+impl From<Stop> for StopPoint {
+    fn from(stop: Stop) -> StopPoint {
         let id = stop.id;
+        if stop.name.is_empty() {
+            warn!("stop_id: {}: for plateform stop_name is required", id);
+        }
+        let coord = Coord::from((stop.lon, stop.lat));
+        if coord == Coord::default() {
+            warn!("stop_id: {}: for plateform coordinates are required", id);
+        }
         StopPoint {
             id,
             name: stop.name,
             visible: stop.visible,
-            coord: Coord {
-                lon: stop.lon,
-                lat: stop.lat,
-            },
+            coord: coord,
             stop_area_id: stop.parent_station.unwrap(),
             timezone: stop.timezone,
             geometry_id: stop.geometry_id,
             equipment_id: stop.equipment_id,
             fare_zone_id: stop.fare_zone_id,
             zone_id: stop.zone_id,
-            stop_type,
+            stop_type: stop.location_type.into(),
```

Comment:
This is why I was talking about implementing `From` instead of `Into` because they both can be used. You could write this instead.
```suggestion
            stop_type: StopLocationType::from(stop.location_type),
```
The `::from()` notation has one advantage (in my opinion), it is forcing you to tell which type you want to convert to, helps the readibility of the code.